src/hooks/useKeyFactory.js

```diff
@@ -0,0 +1,39 @@
+import { useMemo } from 'preact/hooks';
+
+const KEY_LENGTH = 6;
+
+/**
+ * Create a persistent key factory for plain objects without id.
+ *
+ * @example
+ * ```jsx
+ * function List({ objects }) {
+ *   const getKey = useKeyFactory();
+ *   return (<ol>{
+ *     objects.map(obj => {
+ *       const key = getKey(obj);
+ *       return <li key={key}>obj.name</li>
+ *     })
+ *   }</ol>);
+ * }
+ * ```
+ *
+ * @param {any[]} dependencies
+ * @returns {(element: object) => string}
+ */
+export function useKeyFactory(dependencies = []) {
+  const map = useMemo(() => new WeakMap(), dependencies);
```

Comment:
Seems like [we're good](https://caniuse.com/?search=WeakMap) to use `WeakMap` here, only older versions of our supported browsers have problems here.

Reply:
I think weak maps are okay here because we don't expect these maps to have a long life span, right? Did you consider any other types of maps to store the information?

Reply:
I think I will switch to a `Map` so that we can also handle primitive values.

Reply:
Done.